File: 923-3sum-with-multiplicity/923-3sum-with-multiplicity.py
class Solution:
    def threeSumMulti(self, arr: List[int], target: int) -> int:
        ans = 0
        MOD = 1000000007
        
        for i in range(len(arr)):
            _map = defaultdict(int)
            new_target = target - arr[i]

            for j in range(i+1, len(arr)):
                if new_target - arr[j] in _map:
                    ans += _map[new_target - arr[j]]
                    ans %= MOD
                _map[arr[j]] += 1
            
        return ans

File: 923-3sum-with-multiplicity/923-3sum-with-multiplicity.cpp
class Solution {
public:
    int threeSumMulti(vector<int>& arr, int target) {
        //fix on of the numbers
        // do a two sum on the remainin ones
        
        long ans{};
        
        for(auto i=0; i < arr.size(); i ++){
            int elementsCount[101]{};
            int newTarget{target - arr[i]};
            
            for(auto j = i + 1; j  < arr.size(); j++){
                int a = newTarget - arr[j];
                if( 0 <= a and a < 101){
                    ans += elementsCount[a];
                }
                elementsCount[arr[j]] += 1;
            }
            
        }
        
        return ans % (static_cast<int>(pow(10, 9)) + 7);;
        
    }
};